man/ctags-client-tools.7.rst.in

```diff
@@ -343,8 +347,92 @@ Client tools could split the line using the following steps:
+easy: you simply go to that line.
+
+The search pattern resembles an EX command, but as we'll see later, it's
+actually not a valid one, so some manual work are required to process it.
+
+The search pattern could look like ``/pat/``, called "forward search pattern",
+or ``?pat?``, called "backward search pattern". Using a search pattern means
+even if the source file is updated, as long as the part containing the tag
+doesn't change, we could still locate the tag correctly by searching.
+
+When the pattern field only contains the search pattern, you just search for
+it. The search direction (forward/backward) doesn't matter, as it's decided
+solely by whether the ``-B`` option is enabled, and not the actual context. You
+could always start the search from say the beginning of the file.
+
+When both the search pattern and the line number are presented, you could make
+good use of the line number, by going to the line first, then searching for the
+nearest occurence of the pattern. A way to do this is to search both forward
+and backward for the pattern, and when there is a occurence on both sides, go
+to the nearer one.
+
+What's good about this is when there are multiple identical lines in the source
+file (e.g. the COMMON block in Fortran), this could help us find the correct
+one, even after the source file is updated and the tag position is shifted by a
+few lines.
+
+Now let's discuss how to search for the pattern. After you trim the ``/`` or
+``?`` around it, the pattern resembles a regex pattern. It should be a regex
+pattern, as required by being a valid EX command, but it's actually not, as
+you'll see below.
+
+It could begin with a ``^``, which means the pattern starts from the beginning
+of a line. It could also end with an *unescaped* ``$`` which means the pattern
+ends at the end of a line. Let's keep this information, and trim them too.
+
+Now the remaining part is the actual string containing the tag. Some characters
+are escaped:
+
+* ``\``.
+* ``$``, but only at the end of the string.
+* ``/``, but only in forward search patterns.
+* ``?``, but only in backward search patterns.
+
+You need to unescape these to get the literal string. Now you could convert
+this literal string to a regexp that matches it (by escaping, like
+``re.escape`` in Python or ``regexp-quote`` in Elisp), and assemble it with
+``^`` or ``$`` if the pattern originally has it, and finally search for the tag
+using this regexp.
+
+Remark: About a Previous Format of the Pattern Field
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In some earlier versions of @CTAGS_NAME_EXECUTABLE@, the line number in the
+pattern field is the actual line number minus one, for forward search patterns;
+or plus one, for backward search patterns. The idea is to resemble an EX
+command: you go to the line, then search forward/backward for the pattern, and
+you can always find the correct one. But the developers found this denies the
```

Comment:
I guess "the developers" means "the developers of client tools".
Am I correct?

Reply:
No, I mean "developers of ctags". I guess what you are thinking is "using the pattern field this way denies the purpose of the search pattern", and that's a client tool thing, but I'm talking about "the idea of shifting the line number by one is for this usage", so that's a ctags thing.

Reply:
Oh, I see.

Reply:
I changed the sentence to "But this denies...". It's simpler and doesn't affect the meaning.